mithril-aggregator/src/services/snapshotter/compressed_archive_snapshotter.rs

```diff
@@ -114,6 +114,8 @@ impl CompressedArchiveSnapshotter {
     }
 
     fn snapshot<T: TarAppender>(&self, filepath: &Path, appender: T) -> StdResult<OngoingSnapshot> {
+        let temporary_archive_path =
```

Comment:
Maybe this could be done more elegantly with `with_extension`?

Reply:
I tried to use it but `Path::with_extension` replace the extension by the new one, ("archive.tar.gz" become "archive.tar.tmp").
There is also a `Path::with_added_extension` that just add the extension but it's `unstable`